src/Storages/StorageMergeTree.cpp

```diff
@@ -484,6 +495,90 @@ void StorageMergeTree::alter(
     }
 }
 
+/*
+ * For now, this function is meant to be used when exporting to different formats (i.e, the case where data needs to be re-encoded / serialized)
+ * For the cases where this is not necessary, there are way more optimal ways of doing that, such as hard links implemented by `movePartitionToTable`
+ * */
+void StorageMergeTree::exportPartitionToTable(const PartitionCommand & command, ContextPtr query_context)
+{
+    if (!query_context->getSettingsRef()[Setting::allow_experimental_export_merge_tree_partition])
+    {
+        throw Exception(ErrorCodes::SUPPORT_IS_DISABLED,
+            "Exporting merge tree partition is experimental. Set `allow_experimental_export_merge_tree_partition` to enable it");
+    }
+
+    String dest_database = query_context->resolveDatabase(command.to_database);
+    auto dest_storage = DatabaseCatalog::instance().getTable({dest_database, command.to_table}, query_context);
+
+    if (dest_storage->getStorageID() == this->getStorageID())
+    {
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Exporting to the same table is not allowed");
+    }
+
+    if (!dest_storage->supportsImportMergeTreePartition())
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Destination storage {} does not support importing merge tree partitions", dest_storage->getName());
+
+    auto query_to_string = [] (const ASTPtr & ast)
+    {
+        return ast ? ast->formatWithSecretsOneLine() : "";
+    };
+
+    auto src_snapshot = getInMemoryMetadataPtr();
+    auto destination_snapshot = dest_storage->getInMemoryMetadataPtr();
+
+    if (query_to_string(src_snapshot->getPartitionKeyAST()) != query_to_string(destination_snapshot->getPartitionKeyAST()))
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Tables have different partition key");
+
+    String partition_id = getPartitionIDFromQuery(command.partition, getContext());
+
+    auto lock1 = lockForShare(
+        query_context->getCurrentQueryId(),
+        query_context->getSettingsRef()[Setting::lock_acquire_timeout]);
+    auto merges_blocker = stopMergesAndWaitForPartition(partition_id);
+
+    auto all_parts = getVisibleDataPartsVectorInPartition(getContext(), partition_id);
+
+    if (all_parts.empty())
+    {
+        LOG_INFO(log, "No parts to export for partition {}, skipping", partition_id);
+        return;
+    }
+
+    std::lock_guard lock(currently_processing_in_background_mutex);
+
+    if (!already_exported_partition_ids.emplace(partition_id).second)
+    {
+        throw Exception(ErrorCodes::PART_IS_LOCKED, "Partition {} has already been exported", partition_id);
```

Comment:
option to reexport after changes?

Reply:
Well, while you were out we established a partition could not be exported more than once